examples/transit_vnet_common/variables.tf

```
@@ -173,3 +173,67 @@ variable "outbound_lb_name" {
   default     = "lb_outbound"
   type        = string
 }
+
+###### Spoke deployment ########
+variable "spoke_resource_group_name" {
+  description = "Name of the Resource Group to create. If not provided, it will be auto-generated."
+  default     = ""
+  type        = string
+}
+
+variable "spoke_virtual_network_name" {
+  description = "The name of the spoke VNet to create."
+  default     = "spoke_vnet"
+  type        = string
+}
+
+variable "spoke_address_space" {
+  description = "The address space used by the spoke virtual network. You can supply more than one address space."
+  default     = ["10.113.0.0/16"]
+  type        = list(string)
+}
+
+variable "spoke_route_tables" {
+  description = "Definition of Route Tables to create. Refer to the `VNet` module documentation for more information."
+  type        = map(any)
+}
+
+variable "spoke_subnets" {
+  description = "Definition of Subnets to create. Refer to the `VNet` module documentation for more information."
+  type        = map(any)
+}
+
+variable "spoke_vm_offer" {
+  description = "Specifies the offer of the image from the Azure Marketplace"
+  type        = string
+  default     = "UbuntuServer"
+}
+
+variable "spoke_vm_size" {
+  description = "Spoke VM size (type) to be created."
+  type        = string
+  default     = "Standard_DS1_v2"
+}
+
+variable "spoke_vm_name" {
+  description = "The VM name in Spoke Vnet."
+  type        = string
+}
+
+variable "spoke_nb_public_ip" {
+  description = "Number of public IPs to assign corresponding to one IP per vm. Set to 0 to not assign any public IP addresses."
```

Microsoft's comment is not really helpful or I can't understand it all to well. Maybe:

```suggestion
  description = "Set to 0 to not assign any public IP addresses to a spoke Virtual Machine. Set to 1 to create a new Public IP and assign it."
```